### DISCUSSION TITLE
How to calculate misorientation factor m=cosα*cosβ in the two neighboring grains?

### DISCUSSION THREAD

**Discussion by LuqingCui** [mtex-toolbox/mtex#622] [Ask Anything]
Hi all,

I have the EBSD data of an FCC metal. I want to calculate misorientation factor **m=cosα*cosβ** in the two neighboring grains. Where **α** is the angular difference of the closest {111} slip planes in the two neighboring grains and **β** is the angular difference of the closest 〈110〉 slip directions on these two slip planes. 
For example, the Euler angles of the two neighboring grains are (312.1°, 45°, 63.8°) and (290°, 25°, 135°), respectively. Is it possible to calculate the value of **m**?

Looking forward to your reply!
Best regards,
Luqing Cui

**Comment by kilir** [comment]
Hi,
this looks like m', right? Have a look here: https://mtex-toolbox.github.io/slipSystem.mPrime.html
I see the documentation of this is at the moment slighly obscure, the old one still had a demo for slip transmission (based on taking those slip systems with the largest schmid factor), but I think you can easily adopt this by defining your slip system, symmetrise them, rotate them according to your orientations and taking those sets of closest slip planes, closest slip directions.

Hope that helps for the moment.

Cheers,
Rüdiger

For example:
```
cs = crystalSymmetry('m3m')
o = orientation.byEuler([312.1, 45, 63.8;290, 25, 135]*degree,cs)

sS = symmetrise(slipSystem.fcc(cs),'antipodal')

% rotate slip systems
sSo = o*sS;

% smallest angle between slip planes
mang = angle(sSo(1,:).n,sSo(2,:).n);

% select slip systems
% maybe prone to rounding errors, but should work for now
alpha = min(mang);
sSoS = sSo(:,mang - alpha <= 1e-7)

%angle between b
beta = min(angle(sSoS(1,:).b,sSoS(2,:).b))

%m
m = cos(alpha)*cos(beta)
```

**Reply by FCCalloy** [reply]
Hi,
How can the above code be applied to specific EBSD data to calculate the alpha and beta angles of the activated slip planes? Could you give me a demo code of this. I would really appreciate it if you could give me some guidance or explanations on this.
Thank you very much!
Best regards,
Xianghui Zhu

**Comment by LuqingCui** [comment]
Hi Rüdiger，

Yes, I want to calculate m' between two neighboring grains using the corresponding Euler angles, but I don't know how to achieve this using Matlab code. Could you give me a demo code of this?

I have another question: I want to plot the map of the distribution of m' value on the grain boundaries, like follows:
![image](https://user-images.githubusercontent.com/72967420/101479459-849e4b00-3952-11eb-8d08-9774536495e3.png)

The code is as follows:
import_wizard
ebsd=ebsd('ni')
% reconstruct grains
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',15*degree);
% remove small grains
ebsd(grains(grains.diameter<=5)) = [];
% redo grain reconstruction
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',15*degree);
% smooth grain boundaries
grains = smooth(grains,5);
% extract inner phase grain boundaries
gB = grains.boundary('indexed');
plot(grains,grains.meanOrientation,'micronbar','off','linewidth',2)
%We want to consider the following slip systems
sS = slipSystem.fcc(ebsd.CS)
sS = sS.symmetrise;

% rotate slip systems into specimen coordinates
sSLocal = grains.meanOrientation * sS
% compute Schmid factor
sigma = stressTensor.uniaxial(vector3d.Z)
SF = sSLocal.SchmidFactor(sigma);

% take the maxium allong the rows
[SFMax,active] = max(SF,[],2);

% plot the maximum Schmid factor
plot(grains,SFMax,'micronbar','off','linewidth',2)
mtexColorbar location southoutside
% Next we want to visualize the active slip systems.
% take the active slip system and rotate it in specimen coordinates
sSactive = grains.meanOrientation .* sS(active);

hold on
% visualize the trace of the slip plane
quiver(grains,sSactive.trace,'color','b')

% and the slip direction
quiver(grains,sSactive.b,'color','r')
hold off

% We may also analyse the distribution of the slip directions in a pole figure plot
% The same as a contour plot. We see a clear trend towards the east.
plot(sSactive.b,'contourf')

% Compatibility of slip systems
% Next, we want to analyze, how much geometrically compatible the slip systems with highest Schmid factor
% are at the grain boundaries
% some background
plot(grains,'FaceColor',0.8*[1 1 1],'figSize','large')
% compute m'
id = gB.grainId;
**mP = mPrime(sSactive(id(:,1)),sSactive(id(:,2)));**
% plot m' along the grain boundaries
hold on
plot(gB,mP,'linewidth',3)
mtexColorbar
% and plot then the slip trace and the slip direction into the centers of the grains
hold on
quiver(grains,cross(sSactive.n,zvector),'displayName','slip plane')
hold on
quiver(grains,sSactive.b,'displayName','slip direction')
hold off

But when it runs to the line of "mP = mPrime(sSactive(id(:,1)),sSactive(id(:,2)));", it occurs a problem as follows:
![image](https://user-images.githubusercontent.com/72967420/101479734-f6769480-3952-11eb-89e7-9be360c59ea7.png)
Could you tell me how to modify the code?
Thanks a lot in advance.
Best regards,
Luqing Cui

**Comment by kilir** [comment]
Hi,
 your code works fine for me with some example data. Can you try it e.g. with the ferrite dataset? 
`mtexdata ferrite`

Cheers,
Rüdiger

**Reply by Mustafa-Rifat** [reply]
Hello, I am working on the same problem now. In my data, I could plot the m' map like shown in the demo. However, I have 2 questions regarding this.
1.  Let's say we are only interested in getting the m' map. Is it necessary to calculate Schmid Factor to get that? I don't think so. Please correct me if I am wrong.
2. In the [demo page](https://mtex-toolbox.github.io/HomepageOld/files/doc/SlipTransmition.html) it is written "Next, we want to analyze, how much geometrically compatible the slip systems with highest Schmid factor are at the grain boundaries". I am curious how mPrime function knows which slip systems have highest Schmid factor. According to [this page](https://mtex-toolbox.github.io/slipSystem.mPrime.html) explaining the function, it just takes two slip systems corresponding to two neighboring grains as input. Without even calculating SF, how does it know slip systems with highest SF?
